lookout/style/format/benchmarks/evaluate_smoke.py

```diff
@@ -1,48 +1,190 @@
-        for action, _, _, _, _ in matcher.get_opcodes():
-            if action == "equal":
-                continue
-            if action in actions:
-                err += 1
-        return err
+EMPTY = "␣"
+
+
+def align(seq1: Sequence, seq2: Sequence, seq3: Sequence = None
+          ) -> Union[Tuple[Sequence, Sequence],
+                     Tuple[Sequence, Sequence, Sequence]]:
+    """
+    Align two sequences using levenshtein distance.
+
+    For example:
+    In[1]: align("aabc", "abbcc")
+    Out[1]: ("aab␣c␣",
+             "␣abbcc")
+
+    :param seq1: First sequence to align.
+    :param seq2: Second sequence to align.
+    :param seq3: All changes for the second sequence can be applied to seq3. \
+                Used by align3 function.
+
+    :return: Aligned sequences and seq3 modification if specified.
+    """
+    matcher = SequenceMatcher(a=seq1, b=seq2)
+    res1, res2, res3 = [], [], []
+    for action, i, i_end, j, j_end in matcher.get_opcodes():
+        if action == "equal" or action == "replace":
+            res1.append(seq1[i:i_end])
+            res2.append(seq2[j:j_end])
+            if seq3:
+                res3.append(seq3[j:j_end])
+            if i_end - i < j_end - j:
+                res1.append(EMPTY * (j_end - j - i_end + i))
+            elif i_end - i > j_end - j:
+                empty = EMPTY * (i_end - i - j_end + j)
+                res2.append(empty)
+                if seq3:
+                    res3.append(empty)
+        if action == "insert":
+            res1.append(EMPTY * (j_end - j))
+            res2.append(seq2[j:j_end])
+            if seq3:
+                res3.append(seq3[j:j_end])
+        if action == "delete":
+            res1.append(seq1[i:i_end])
+            empty = EMPTY * (i_end - i)
+            res2.append(empty)
+            if seq3:
+                res3.append(empty)
+    if seq3:
+        return "".join(res1), "".join(res2), "".join(res3)
+    return "".join(res1), "".join(res2)
+
+
+def align3(seq1: Sequence, seq2: Sequence, seq3: Sequence) -> Tuple[Sequence, Sequence, Sequence]:
+    """
+    Align three sequences using levenshtein distance.
+
+    For example:
+    In[1]: align("aabc", "abbcc", "ccdd")
+    Out[1]: ("aab␣c␣␣␣",
+             "␣abbcc␣␣",
+             "␣␣␣␣ccdd")
+
+    It can be suboptimal because heuristic is used because true calculation requires
+    ~ len(seq1) * len(seq2) * len(seq3) time.
+
+    :param seq1: First sequence to align.
+    :param seq2: Second sequence to align.
+    :param seq3: Third sequence to align.
+    :return: Aligned sequences
+    """
+    aseq1, aseq2 = align(seq1, seq2)
+    res3, res1, res2 = align(seq3, aseq1, aseq2)
+    return res1, res2, res3
 
-    return edit_distance(predicted_lines, correct_lines) / \
-        edit_distance(head_lines, correct_lines)
+
+def metrics(init: str, correct: str, model_out: str) -> Tuple[int, int, int, int]:
+    """
+    Calculate quality metrics for aligned sequences.
```

Comment:
metrics can be calculated for models, not for sequences. You **use** aligned sequences to calc metrics as far as I see

Reply:
A metric can be calculated between any two points in metric space. 

What do you want me to change?

`def calc_metrics`?